modules/openapi-generator/src/main/resources/python/model_utils.mustache

```diff
@@ -1450,7 +1450,7 @@ def get_allof_instances(self, model_args, constant_args):
     for allof_class in self._composed_schemas['allOf']:
 
         try:
-            allof_instance = allof_class(**model_args, **constant_args)
+            allof_instance = allof_class._new_from_openapi_data(**model_args, **constant_args)
```

Comment:
_new_from_openapi_data assumes that the data is coming from the server which is not always true. Sometimes it is coming from the client and sometimes it is coming from the server.
How about doing what I suggest here: https://github.com/OpenAPITools/openapi-generator/pull/9735#discussion_r658883937
- When coming from the server use `allof_instance = allof_class._from_openapi_data(*args, **kwargs)`
- When coming from the client use `allof_instance = allof_class._new_from_openapi_data(**model_args, **constant_args)`
- One can use `kwargs['_spec_property_naming']` to determine if the payload is coming from the server or client, _spec_property_naming=True when coming from the server and false otherwise